test/extensions/filters/network/ext_authz/ext_authz_fuzz_test.cc

```diff
@@ -62,14 +62,32 @@ DEFINE_PROTO_FUZZER(const envoy::extensions::filters::network::ext_authz::ExtAut
   }
 
   Stats::TestUtil::TestStore stats_store;
-  Filters::Common::ExtAuthz::MockClient* client = new Filters::Common::ExtAuthz::MockClient();
   envoy::extensions::filters::network::ext_authz::v3::ExtAuthz proto_config = input.config();
   NiceMock<Server::Configuration::MockServerFactoryContext> context;
 
-  ConfigSharedPtr config =
-      std::make_shared<Config>(proto_config, *stats_store.rootScope(), context);
-  std::unique_ptr<Filter> filter =
-      std::make_unique<Filter>(config, Filters::Common::ExtAuthz::ClientPtr{client});
+  ConfigSharedPtr config;
+  try {
+    config = std::make_shared<Config>(proto_config, *stats_store.rootScope(), context);
+  } catch (const EnvoyException& e) {
+    ENVOY_LOG_MISC(debug, "EnvoyException during validation: {}", e.what());
+    return;
+  }
+
+  auto client = std::make_unique<Filters::Common::ExtAuthz::MockClient>();
+  for (const auto& action : input.actions()) {
```

Comment:
This is just setting up mock calls on the client since we later transfer client ownership to the filter. This doesn't work as intended, however, because only the last call to WillByDefault will be in effect when onData is actually called.

I don't think it's possible to use only a unique_ptr for the client. We need access to the client after transferring "ownership" to the filter. I've got to add the raw pointer back unfortunately.